garageScript/c0d3-app
# feat: Add next-auth middleware to protect pages

## Motivation

At present, we do not have a streamlined approach to prevent unauthorized access to pages that require the user to be logged in. The current implementation involves manual checks on the session to determine if the user needs to be redirected to the login page or not.

## Solution

Introduce `next-auth` middleware to handle the protection of pages that require the user to be logged in. The middleware will maintain a list of protected paths and in case the user is not logged in, they will be immediately redirected to the `/login` page. This not only simplifies the codebase by removing the redundant code for page protection but also saves bandwidth for the user.

## Changes

- Implement `next-auth` middleware to protect the `/settings/account` page
- Create a constant for the path `/settings/account`
- Set the redirection to `/login` for unauthenticated users
- [Remove the `secret` option in favor of using `NEXTAUTH_SECRET` in Vercel's project environment variables](https://next-auth.js.org/configuration/options#nextauth_secret)
- Add a test file to cover the `middleware.ts` file

## Testing

1. Log out of the application
2. Access the `/settings/account` page and verify if it redirects to `/login`
3. Log in to the application by filling in the login form
4. Verify if it redirects to `/settings/account` after successful login

## Additional Info

In future pull requests, more pages such as `/review/[lessonSlug]` will be added to the list of protected pages.

## Related issues

- #1629
